lib/kubernetes-deploy/kubernetes_resource/persistent_volume_claim.rb

```diff
@@ -3,12 +3,66 @@ module KubernetesDeploy
   class PersistentVolumeClaim < KubernetesResource
     TIMEOUT = 5.minutes
 
+    def init_storage_class(cache)
+      @storage_class = {}
+
+      if @definition.dig("spec", "storageClassName").nil?
+        # if no storage class is defined we try to find the default one
+        # FIXME: This assumes the DefaultStorageClass admission plugin is turned on,
+        # need a way to determine this
+        is_default_class = "storageclass.beta.kubernetes.io/is-default-class"
+
+        default_sc = cache.get_all("StorageClass").select do |sc|
+          sc.dig("metadata", "annotations", is_default_class) == "true"
+        end
+
+        if default_sc.length != 1
+          warn_msg = "Multiple default StorageClasses found. If the DefaultStorageClass " \
+            "admission plugin is turned on, all PVC creation will fail."
+          @logger.warn(warn_msg) if default_sc.length > 1
+          return
+        else
+          # using default storage class
+          sc_name = default_sc[0]["metadata"]["name"]
+        end
+      else
+        # using storage class from pvc definition
+        sc_name = @definition.dig("spec", "storageClassName")
+
+        # a storageClassName of "" is an explicit way of saying you want a PV
+        # with no defined StorageClass. We won't look up a storage_class
```

Comment:
Just to confirm this case is different than using the default storage class?

Reply:
Yeah - I believe this is used to explicitly opt out of using the default storage class (or any storage class):
https://kubernetes.io/docs/concepts/storage/persistent-volumes/#class-1